.github/workflows/build-and-test.yml

```
@@ -0,0 +1,21 @@
+name: Build and Test
+on: [pull_request, push]
+jobs:
+  build-and-test:
+    runs-on: ubuntu-18.04
+    strategy:
+      matrix:
+        include:
+          - DISTRO: "tomcat"
+            EE: "false"
+          - DISTRO: "wildfly"
+            EE: "false"
+          - DISTRO: "run"
+            EE: "false"
+    steps:
+      - uses: actions/checkout@v2
+      - name: Execute pipeline
+        run: ./pipeline.sh
+        env:
+          DISTRO: ${{ matrix.DISTRO }}
+          EE: ${{ matrix.EE }}
```

I'd suggest adding a conditional `run` step here (reason in other comment), which will only get executed for `next` and tags matching the semver pattern:


```suggestion
          EE: "false"
      - name: Publish image
        if: {{ github.ref == "refs/heads/next" || startsWith(github.ref, "refs/tags/v7.") }}
        run: ./release.sh
        env:
          DISTRO: ${{ matrix.DISTRO}}
          EE: "false"
          DOCKER_HUB_PASSWORD: ${{ secrets.DOCKER_HUB_PASSWORD }}
          DOCKER_HUB_USERNAME:  ${{ secrets.DOCKER_HUB_USERNAME }}
```

---

you're right, makes sense! Didn't see prior that it required the image to already been built.